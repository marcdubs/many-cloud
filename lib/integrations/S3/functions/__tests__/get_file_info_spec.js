describe("get_file_info", () => {
  describe("when the operation is successfull", () => {
    const expected_result = "api_result",
      expected_file_id = "1234";
    let data, func, actual;
    beforeEach(async done => {
      data = {
        s3: {
          headObject: jest.fn().mockImplementation((params, callback) => {
            callback(null, expected_result);
          })
        }
      };
      func = require("../get_file_info")(data);
      actual = await func(expected_file_id);
      done();
    });

    it("calls headObject", () => {
      expect(data.s3.headObject).toHaveBeenCalled();
    });

    it("calls headObject with fileID", () => {
      expect(data.s3.headObject.mock.calls[0][0].Key).toEqual(expected_file_id);
    });
  });

  describe("when the operation is unsuccessfull", () => {
    const expected_error = "Oh boy!";
    let data, func, actual;
    beforeEach(() => {
      data = {
        s3: {
          headObject: jest.fn().mockImplementation((params, callback) => {
            callback(new Error(expected_error), null);
          })
        }
      };
      func = require("../get_file_info")(data);
    });

    it("rejects with the error", async () => {
      await expect(func()).rejects.toMatchSnapshot();
    });
  });
});
