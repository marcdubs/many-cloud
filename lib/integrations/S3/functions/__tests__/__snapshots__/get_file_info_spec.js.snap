// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`get_file_info when the operation is unsuccessfull rejects with the error 1`] = `[Error: Oh boy!]`;
